draft-ietf-quic-transport.md

```diff
@@ -2740,7 +2697,57 @@ that it sends.
 Strategies and implications of the frequency of generating acknowledgments are
 discussed in more detail in {{QUIC-RECOVERY}}.
 
-## Special Considerations for PMTU Discovery
+
+## Packet Size
+
+Clients MUST ensure that the first packet in a connection, and any
+retransmissions of those octets, has a QUIC packet size of least 1200 octets.
+The packet size for a QUIC packet includes the QUIC header and integrity check,
+but not the UDP or IP header.
+
+The initial client packet SHOULD be padded to exactly 1200 octets unless the
+client has a reasonable assurance that the Path Maximum Transmission Unit (PMTU)
+is larger.  Sending a packet of this size ensures that the network path supports
+an MTU of this size and helps reduce the amplitude of amplification attacks
+caused by server responses toward an unverified client address.
+
+Servers MUST ignore an initial plaintext packet from a client if its total size
+is less than 1200 octets.
+
+
+## Path Maximum Transmission Unit
+
+The Path Maximum Transmission Unit (PMTU) is the maximum size of the entire IP
+header, UDP header, and UDP payload. The UDP payload includes the QUIC packet
+header, protected payload, and any authentication fields.
+
+All QUIC packets SHOULD be sized to fit within the estimated PMTU to avoid IP
+fragmentation or packet drops. To optimize bandwidth efficiency, endpoints
+SHOULD use Packetization Layer PMTU Discovery ({{!PLPMTUD=RFC4821}}) and MAY use
+PMTU Discovery ({{!PMTUDv4=RFC1191}}, {{!PMTUDv6=RFC8201}}) for detecting the
+PMTU, setting the PMTU appropriately, and storing the result of previous PMTU
+determinations.
+
+In the absence of these mechanisms, QUIC endpoints SHOULD NOT send IP packets
+larger than 1280 octets. Assuming the minimum IP header size, this results in
+a QUIC packet size of 1232 octets for IPv6 and 1252 octets for IPv4.
+
+QUIC endpoints that implement any kind of PMTU discovery SHOULD maintain an
+estimate for each combination of local and remote IP addresses (as each pairing
+could have a different maximum MTU in the path).
+
+QUIC depends on the network path supporting a MTU of at least 1280 octets. This
+is the IPv6 minimum and therefore also supported by most modern IPv4 networks.
+An endpoint MUST NOT reduce their MTU below this number, even if it receives
+signals that indicate a smaller limit might exist.
+
+If a QUIC endpoint determines that the PMTU between any pair of local and remote
```

Comment:
There's obviously a good reason for this, which I can't seem to remember. Why do we disallow smaller packets? Can you add a note here about that as well, so it's obvious to any reader?

Reply:
My reading of this is that the maximum must not be reduced, but if you have less to send, go ahead. This will trigger packet loss and eventual connection failure when sending bulk data on a network that does not honor the 1200 minimum and therefore possibly trigger a connection migration if possible, or otherwise stop a hopeless quest.

Reply:
Another reason may be to prevent DoS attacks by injection of unverifiable network congestion control / status data.

Reply:
I'm going to request that you open an issue if you care about this.  This is old text and I'm not really equipped to deal with it right now.  I *think* that @mikkelfj is approximately right.  It might also be to avoid using paths that will drop random packets.

Frankly, I don't think that we need the requirement.  If you *know* that this has happened, you can deal.  The only real cost at that point is that you might be unable to create a new connection on that path.